auth.go

```diff
@@ -474,32 +435,87 @@ func (x *Central) Login(identity, password string) (sessionkey string, token *To
 }
 
 // Merges ldap with user store every 10 seconds
-func (x *Central) Merge() error {
+func (x *Central) MergeTicker() error {
 	x.Log.Info("Starting LDAP merge process")
-	x.Ticker = time.NewTicker(time.Second * 10)
+	x.Ticker = time.NewTicker(ldapMergeTickerSeconds)
 	go func() {
-		for _ = range x.Ticker.C {
+		mergeTick := func() error {
 			x.Log.Infof("Merging at (%v)", time.Now().UnixNano()/int64(time.Millisecond))
-			ldapUsers, err := x.userStore.GetLdapUsers()
-			//for _,ldapUser := range ldapUsers {
-			//	x.Log.Infof("User: %v", ldapUser.Username)
-			//}
+			ldapUsers, err := x.ldap.GetLdapUsers()
 			if err != nil {
-				x.Log.Errorf("Failed to retrieve users from LDAP server (%v)", err)
-				break
+				x.Log.Errorf("Failed to retrieve users from LDAP server for merge to take place (%v)", err)
+				return err
 			}
-			err = x.userStore.Merge(ldapUsers)
+			imqsUsers, err := x.userStore.GetIdentities()
 			if err != nil {
+				x.Log.Errorf("Failed to retrieve users from Userstore for merge to take place (%v)", err)
+				return err
+			}
+			if err = x.Merge(ldapUsers, imqsUsers); err != nil {
 				x.Log.Errorf("Failed to merge LDAP users (%v)", err)
-				break
+				return err
 			}
 			x.Log.Infof("Merging complete at (%v)", time.Now().UnixNano()/int64(time.Millisecond))
+			return nil
+		}
+		if err := mergeTick(); err == nil {
+			for _ = range x.Ticker.C {
+				if err := mergeTick(); err != nil {
+					break
+				}
+			}
+			x.Ticker.Stop()
+
 		}
-		x.Ticker.Stop()
 	}()
 
 	return nil
+}
+
+// Merges the ldap users and imqs users
+func (x *Central) Merge(ldapUsers []AuthUser, imqsUsers []AuthUser) error {
+	found := false
+	// Insert or update
+	for _, ldapUser := range ldapUsers {
+		for _, imqsUser := range imqsUsers {
+			if ldapUser.Username == imqsUser.Username {
```

Comment:
You need to create maps here, where the keys are the usernames, to avoid the N^2 execution time. If a client has 1000 LDAP users (a very reasonable expectation), then we have 2 million string comparisons to do. After implementing this change, it will be even clearer why `found` needs to be more tightly scoped, because you'll write a single line:
```go
imqsUser, found := usernameToImqs[ldapUser.Username]
```